src/view/configuration/utils/getPartsFromEnvironmentCompositeId.js

```diff
@@ -10,22 +10,18 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-let abortController;
-
 /**
- * This is shared amongst schema selection async processes to
- * ensure that if a user selects a new sandbox or schema, that
- * prior async processes will be appropriately canceled in order
- * to avoid race conditions.
+ * Returns the two parts of a composite ID: the config ID
+ * and the environment name. If the composite ID isn't actually
+ * a composite ID, the returned environment name will be undefined.
  */
-const abortPreviousRequestsAndCreateNewSignal = () => {
-  if (abortController) {
-    abortController.abort();
-  }
-
-  abortController = new AbortController();
+const getPartsFromEnvironmentCompositeId = compositeId => {
+  const [edgeConfigId, environmentName] = compositeId.split(":");
 
-  return abortController.signal;
+  return {
+    edgeConfigId,
+    environmentName
+  };
 };
 
-export default abortPreviousRequestsAndCreateNewSignal;
```

Comment:
This diff is messed up; this wasn't a renamed file. `abortPreviousRequestsAndCreateNewSignal` was something I was using canceling requests in the XDM object data element. Notice that I had a "global" `abortController`, which was fine for the XDM object case, but it meant I couldn't re-use it for the  edge config selection in the configuration view. I decided to change it to a re-usable React hook, so the abortController would be scoped to the component using it. See `useAbortPreviousRequestsAndCreateSignal.jsx`.